azkaban-common/src/main/java/azkaban/executor/AbstractExecutorManagerAdapter.java

```diff
@@ -438,8 +460,18 @@ protected LogData getFlowLogData(final ExecutableFlow exFlow, final int offset,
               typeParam, offsetParam, lengthParam);
       return LogData.createLogDataFromObject(result);
     } else {
-      return this.executorLoader.fetchLogs(exFlow.getExecutionId(), "", 0, offset,
-          length);
+      LogData logData = this.nearlineExecutionLogsLoader.fetchLogs(exFlow.getExecutionId(), "", 0,
+          offset, length);
+
+      // Return offline logs if nearline logs are empty or the flow in kubernetes pod crashed
+      if (offlineLogsLoaderEnabled && offlineExecutionLogsLoader.isPresent() &&
+          (logData.getLength() == 0 ||
+              (exFlow.getDispatchMethod() == DispatchMethod.CONTAINERIZED &&
```

Comment:
do execution_stopped and killed flows have logdata length > 0? If not, the second condition only needs to check the logdata length.

Reply:
do execution_stopped and killed flows have logdata length > 0?: yes. But Adam's LogPodDetails PR will write additional info into mysql if crashed. For safety, I just add the second condition here. If we finally decide not to add the LogPodDetails, I can remove the second condition later.